### ISSUE TITLE
Fail to parse some names?

### ISSUE THREAD

**Issue by leungmanhin** [opencog/opencog#1907]
When parsing a sentence `My name is Raymond Reddington.`, there is a ConceptNode with an empty string in the R2L outputs:
```
      (InheritanceLink
         (ConceptNode "Reddington@c039f8e3-935d-4b4a-834a-93d84ee0f586")
         (ConceptNode "Raymond_Reddington" (ptv 1 0 1))
      )
      (InheritanceLink
         (ConceptNode "name@54089fcd-d243-4304-b8fa-b330940d79ed")
         (ConceptNode "Reddington@c039f8e3-935d-4b4a-834a-93d84ee0f586")
      )
      (InheritanceLink
         (ConceptNode "Raymond@7f806895-55f2-427a-8321-e2fcb0c48110")
         (ConceptNode "" (ptv 1 0 1))
      )
```

**Comment by linas** [comment]
running from master I get `ERROR: Unbound variable: pre-be-inheritance-rule`  ...

**Comment by linas** [comment]
It has something to do with the fact that relex mashes together "Raymond_Reddington" into one word. 

As a side-effect the following is generated by relex itself:
```
(LemmaLink (stv 1.0 1.0)
   (WordInstanceNode "Raymond@f597ff94-fc19-452d-af24-1d1058e8d4a3")
   (WordNode "")
)
(LemmaLink (stv 1.0 1.0)
   (WordInstanceNode "Reddington@eb836e90-7ba7-4140-a779-4be9506e8fbb")
   (WordNode "Raymond_Reddington")
)
```

Then R2L converts that lemma-link into an inheritancelink thus leading to the break.

I think I know an ugly hack to fix this ...

**Comment by linas** [comment]
How can one get a list of all the rules that were applied? One of rules is converting the Lemma link like this but I can't tell which one.

**Comment by AmeBel** [comment]
Rule selection isn't performed in r2l, that is all the rules are run and if the implicant of the rules are satisfied in the focus-set (the relex-parse output ) then they are grounded, thus there is no way of knowing which rules are applied presently, AFAIK.

**Comment by leungmanhin** [comment]
Oh there is a log file called `forward_chainer.log` in `opencog/build` that contains info about which rule FC was applying and the result

**Comment by linas** [comment]
The forward_chainer.log does not contain
```
      (InheritanceLink
         (ConceptNode "Raymond@7f806895-55f2-427a-8321-e2fcb0c48110")
         (ConceptNode "" (ptv 1 0 1))
      )
```

which suggests that it is not the one creating the above.  I cannot find what is creating the above, though.

**Comment by leungmanhin** [comment]
Looks like it is the definite rule:
```
[2015-12-02 03:53:18:962] [DEBUG] Applying rule in focus set (BindLink
  (VariableList
    (TypedVariableLink
      (VariableNode "$a-parse") ; [6451][6450]
      (TypeNode "ParseNode") ; [6452][6450]
    ) ; [6453][6450]
    (TypedVariableLink
      (VariableNode "$noun") ; [6454][6450]
      (TypeNode "WordInstanceNode") ; [6455][6450]
    ) ; [6456][6450]
    (TypedVariableLink
      (VariableNode "$lemma") ; [6457][6450]
      (TypeNode "WordNode") ; [6458][6450]
    ) ; [6459][6450]
  ) ; [6460][6450]
  (AndLink
    (WordInstanceLink
      (VariableNode "$noun") ; [6454][6450]
      (VariableNode "$a-parse") ; [6451][6450]
    ) ; [6461][6450]
    (LemmaLink
      (VariableNode "$noun") ; [6454][6450]
      (VariableNode "$lemma") ; [6457][6450]
    ) ; [6462][6450]
    (InheritanceLink
      (VariableNode "$noun") ; [6454][6450]
      (DefinedLinguisticConceptNode "definite") ; [6193][6021]
    ) ; [6463][6450]
  ) ; [6464][6450]
  (ExecutionOutputLink
    (GroundedSchemaNode "scm: definite-rule") ; [6465][6450]
    (ListLink
      (VariableNode "$lemma") ; [6457][6450]
      (VariableNode "$noun") ; [6454][6450]
    ) ; [6466][6450]
  ) ; [6467][6450]
) ; [6468][6450]
 
[2015-12-02 03:53:18:964] [DEBUG] Result is (SetLink
  (ListLink
    (InheritanceLink
      (ConceptNode "Raymond@1d073b35-6abc-4698-a649-13b4723084f4") ; [6471][1]
      (ConceptNode "#4695") ; [4695][1]
    ) ; [6472][1]
    (ReferenceLink
      (ConceptNode "Raymond@1d073b35-6abc-4698-a649-13b4723084f4") ; [6471][1]
      (WordInstanceNode "Raymond@1d073b35-6abc-4698-a649-13b4723084f4") ; [5915][1]
    ) ; [6473][1]
    (EvaluationLink
      (DefinedLinguisticPredicateNode "definite") ; [1292][1]
      (ListLink
        (ConceptNode "Raymond@1d073b35-6abc-4698-a649-13b4723084f4") ; [6471][1]
      ) ; [6474][1]
    ) ; [6475][1]
  ) ; [6476][1]
  (ListLink
    (InheritanceLink
      (ConceptNode "Reddington@27fee2fb-087d-4a52-af0c-c410d3a435c6") ; [6365][1]
      (ConceptNode "Raymond_Reddington") ; [6368][1]
    ) ; [6369][1]
    (ReferenceLink
      (ConceptNode "Reddington@27fee2fb-087d-4a52-af0c-c410d3a435c6") ; [6365][1]
      (WordInstanceNode "Reddington@27fee2fb-087d-4a52-af0c-c410d3a435c6") ; [5921][1]
    ) ; [6366][1]
    (EvaluationLink
      (DefinedLinguisticPredicateNode "definite") ; [1292][1]
      (ListLink
        (ConceptNode "Reddington@27fee2fb-087d-4a52-af0c-c410d3a435c6") ; [6365][1]
      ) ; [6478][1]
    ) ; [6479][1]
  ) ; [6480][1]
  (ListLink
    (InheritanceLink
      (ConceptNode "name@71d1c7cb-d87b-462c-b155-f847d5baa30b") ; [6363][1]
      (ConceptNode "name") ; [4581][1]
    ) ; [6367][1]
    (ReferenceLink
      (ConceptNode "name@71d1c7cb-d87b-462c-b155-f847d5baa30b") ; [6363][1]
      (WordInstanceNode "name@71d1c7cb-d87b-462c-b155-f847d5baa30b") ; [5905][1]
    ) ; [6364][1]
    (EvaluationLink
      (DefinedLinguisticPredicateNode "definite") ; [1292][1]
      (ListLink
        (ConceptNode "name@71d1c7cb-d87b-462c-b155-f847d5baa30b") ; [6363][1]
      ) ; [6482][1]
    ) ; [6483][1]
  ) ; [6484][1]
  (ListLink
    (InheritanceLink
      (ConceptNode "my@3169b81a-8d66-46b1-914c-d0e8bca5e078") ; [6486][1]
      (ConceptNode "me") ; [4687][1]
    ) ; [6487][1]
    (ReferenceLink
      (ConceptNode "my@3169b81a-8d66-46b1-914c-d0e8bca5e078") ; [6486][1]
      (WordInstanceNode "my@3169b81a-8d66-46b1-914c-d0e8bca5e078") ; [5900][1]
    ) ; [6488][1]
    (EvaluationLink
      (DefinedLinguisticPredicateNode "definite") ; [1292][1]
      (ListLink
        (ConceptNode "my@3169b81a-8d66-46b1-914c-d0e8bca5e078") ; [6486][1]
      ) ; [6489][1]
    ) ; [6490][1]
  ) ; [6491][1]
) ; [6492][1]
```

**Comment by linas** [comment]
closing, fixed correctly in opencog/relex#243
